### PR TITLE
[EAGLE-918] Support auth annotations PermitAll/DenyAll/RolesAllowed and API

### PR DESCRIPTION
**Pull request by haoch**
* Improve eagle auth framework to support annotations under Jersey 1.9 and Dropwizard 0.7.1
    
        @PermitAll
        @DenyAll
        @RolesAllowed
        @Auth

* Add Authentication API

        GET /auth/principal
        GET /auth/validate
        GET /auth/login

* Add password encryption and EncrytTool

        java ServerMain encrypt [text to encrypt]

### CODE REVIEW

**Comment by haoch** [comment]
@zombieJ please move on to refactor eagle login page after this PR merged.

File: eagle-server/src/main/java/org/apache/eagle/server/authentication/config/UserAccount.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
```
Comment (r7raul1984) [inline eagle-server/src/main/java/org/apache/eagle/server/authentication/config/UserAccount.java]: In order to make it more safe，I think that we should encrypt  the password .

Comment (haoch) [inline eagle-server/src/main/java/org/apache/eagle/server/authentication/config/UserAccount.java]: Good point.

Comment (haoch) [inline eagle-server/src/main/java/org/apache/eagle/server/authentication/config/UserAccount.java]: Resolved with password encryption.

**Comment by haoch** [comment]
@r7raul1984 I have finished this PR, please help continue to review. If no more problem, I will merge it after approved.

**Comment by haoch** [comment]
Building failure is caused by checkstyle problem, fixed in  43827444eb0e8fcd46ce6983bf456b920eb5b8b1

**Comment by haoch** [comment]
Merged as Approved by @r7raul1984

**Pull request closed without merging**